test/unittests/state_bloom_test.cpp

```diff
@@ -0,0 +1,64 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../utils/utils.hpp"
+#include <gtest/gtest.h>
+#include <test/state/bloom.hpp>
+#include <test/statetest/statetest.hpp>
+
+TEST(state_bloom, combine_blooms)
+{
+    using namespace evmone::state;
+
+    // https://sepolia.etherscan.io/block/2946594
```

Comment:
Nice idea. Maybe create another test for this single transaction from the same block: https://sepolia.etherscan.io/tx/0xe07740662828a58224e2efedf3accd2763ae52c5b281b120019c3b4a0b6862dd#eventlog